rskj-core/src/test/java/co/rsk/peg/StateForProposedFederatorTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2024 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.peg;
+
+import static co.rsk.peg.PegTestUtils.createHash3;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.crypto.Keccak256;
+import java.util.AbstractMap;
+import java.util.Map;
+import org.ethereum.util.RLP;
+import org.ethereum.util.RLPList;
+import org.junit.jupiter.api.Test;
+
+class StateForProposedFederatorTest {
+
+    private static final NetworkParameters NETWORK_PARAMETERS = NetworkParameters.fromID(NetworkParameters.ID_MAINNET);
+
+    @Test
+    void stateForProposedFederator_whenSerializeAndDeserialize_shouldHaveEqualState() {
```

Comment:
a test when there's no tx waiting for signatures?

Reply:
Updated